pittgoogle/pubsub.py

```diff
@@ -409,8 +434,28 @@ def _create(self) -> google.cloud.pubsub_v1.types.Subscription:
         if self.topic is None:
             raise TypeError("The subscription needs to be created but no topic was provided.")
 
+        if self.udf:
+            import re
+
+            # the function name must match what is defined in the UDF code
+            # we parse through the code using regex to find it
+            match = re.search(r"function\s+([a-zA-Z0-9_]+)\s*\(", self.udf.replace("\n", " "))
+            _function_name = match.group(1) if match else "filter"
+            if not match:
+                LOGGER.warning("Could not parse function name from UDF; using default 'filter'.")
```

Comment:
This isn't necessarily a filter though, right? It could be a transformation that doesn't filter.